Bottom: 26ebf583e7f29345e821756cf1a3545854b5468e
Top:    d8d93974e13ea68f2747ba8976944e98433fde13
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2014-01-02 09:21:03 -0500

Refresh of zarlink

---

diff --git a/package/kernel/linux/modules/sound.mk b/package/kernel/linux/modules/sound.mk
index f3e5673..f7e3e7d 100644
--- a/package/kernel/linux/modules/sound.mk
+++ b/package/kernel/linux/modules/sound.mk
@@ -284,10 +284,14 @@ $(eval $(call KernelPackage,pcspkr))
 define KernelPackage/sound-soc-rt5350-pcm
   TITLE:=RT5350 SoC support
   KCONFIG:=\
+	CONFIG_DMA_RALINK \
+	CONFIG_NET_DMA=n \
 	CONFIG_SND_SOC_RT5350_PCM
   FILES:= \
-	$(LINUX_DIR)/sound/soc/ralink/snd-soc-rt5350-pcm.ko
-  AUTOLOAD:=$(call AutoLoad,56,snd-soc-rt5350-pcm)
+	$(LINUX_DIR)/sound/soc/ralink/snd-soc-rt5350-pcm.ko \
+	$(LINUX_DIR)/drivers/dma/virt-dma.ko \
+	$(LINUX_DIR)/drivers/dma/ralink-gdma.ko
+  AUTOLOAD:=$(call AutoLoad,56,snd-soc-rt5350-pcm ralink-gdma)
   DEPENDS:=@TARGET_ramips +kmod-sound-soc-core
   $(call AddDepends/sound)
 endef
diff --git a/target/linux/ramips/dts/rt5350.dtsi b/target/linux/ramips/dts/rt5350.dtsi
index a68f199..9ada213 100644
--- a/target/linux/ramips/dts/rt5350.dtsi
+++ b/target/linux/ramips/dts/rt5350.dtsi
@@ -213,16 +213,20 @@
 			interrupt-parent = <&intc>;
 			interrupts = <4>;
 
+			dmas = <&gdma 4>;
+			dma-names = "rx-tx";
 			status = "disabled";
 		};
 
-		gdma@2800 {
+		gdma: gdma@2800 {
 			compatible = "ralink,rt5350-gdma", "ralink,rt2880-gdma";
 			reg = <0x2800 0x100>;
 
 			interrupt-parent = <&intc>;
 			interrupts = <7>;
 
+			#dma-cells = <1>;
+			dma-channels = <16>;
 			status = "disabled";
 		};
 	};
diff --git a/target/linux/ramips/patches-3.10/0320-rt5350-zarlink.patch b/target/linux/ramips/patches-3.10/0320-rt5350-zarlink.patch
index 652f868..3c02d74 100644
--- a/target/linux/ramips/patches-3.10/0320-rt5350-zarlink.patch
+++ b/target/linux/ramips/patches-3.10/0320-rt5350-zarlink.patch
@@ -812,15 +812,16 @@
 +#endif
 --- a/sound/soc/ralink/Kconfig
 +++ b/sound/soc/ralink/Kconfig
-@@ -13,3 +13,20 @@ config SND_MT7620_SOC_WM8960
+@@ -13,3 +13,21 @@ config SND_MT7620_SOC_WM8960
  	help
  	  Say Y if you want to add support for ASoC audio on the Qi LB60 board
  	  a.k.a Qi Ben NanoNote.
 +
 +config SND_SOC_RT5350_PCM
 +	depends on SOC_RT305X && SND_SOC
-+	select SND_SOC_GENERIC_DMAENGINE_PCM
++	select DMADEVICES
 +	select DMA_RALINK
++	select SND_SOC_GENERIC_DMAENGINE_PCM
 +	tristate "SoC Audio (PCM protocol) for Ralink RT5350 SoC"
 +	help
 +	  Say Y if you want to use I2S protocol and PCM codec on Ralink RT5350
@@ -1503,12 +1504,24 @@
  
 --- a/drivers/dma/Kconfig
 +++ b/drivers/dma/Kconfig
-@@ -314,7 +314,7 @@ config MMP_PDMA
+@@ -314,9 +314,11 @@ config MMP_PDMA
  
  config DMA_RALINK
  	tristate "RALINK DMA support"
 -	depends on RALINK && SOC_MT7620
-+	depends on RALINK && (SOC_MT7620 | SOC_RT305X)
++	depends on RALINK && (SOC_MT7620 || SOC_RT305X)
  	select DMA_ENGINE
  	select DMA_VIRTUAL_CHANNELS
++	help
++	  Support the GDMA engine for Ralink MIPS platfrom.
  
+ config DMA_ENGINE
+ 	bool
+--- a/drivers/dma/ralink-gdma.c
++++ b/drivers/dma/ralink-gdma.c
+@@ -574,4 +574,4 @@ module_platform_driver(gdma_dma_driver);
+ 
+ MODULE_AUTHOR("Lars-Peter Clausen <lars@metafoo.de>");
+ MODULE_DESCRIPTION("GDMA4740 DMA driver");
+-MODULE_LICENSE("GPLv2");
++MODULE_LICENSE("GPL");
